contexts/SupabaseContext.tsx

```diff
@@ -97,8 +104,8 @@ function SupabaseProvider({ children }: { children: ReactChild | ReactChildren }
 
   let typedUser = null;
   if (user) {
-    if (user.app_metadata.provider === "google") {
-      typedUser = user as GoogleUser;
+    if (isGoogleUser(user)) {
+      typedUser = user;
     } else {
       typedUser = user as EmailUser;
     }
```

Comment:
`typedUser` ends up being typed as `EmailUser | null` after this block, rather than the intended `EmailUser | GoogleUser | null`...not sure why, but it wasn't because I added `isGoogleUser`